pkg/storage/unified/search/user.go

```diff
@@ -0,0 +1,80 @@
+package search
+
+import (
+	"context"
+
+	iamv0 "github.com/grafana/grafana/apps/iam/pkg/apis/iam/v0alpha1"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"github.com/grafana/grafana/pkg/storage/unified/resource"
+	"github.com/grafana/grafana/pkg/storage/unified/resourcepb"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	USER_EMAIL = "email"
+	USER_LOGIN = "login"
+)
+
+func GetUserBuilder() (resource.DocumentBuilderInfo, error) {
+	fields, err := resource.NewSearchableDocumentFields([]*resourcepb.ResourceTableColumnDefinition{
+		{
+			Name:        USER_EMAIL,
+			Type:        resourcepb.ResourceTableColumnDefinition_STRING,
+			Description: "The email address of the user",
+			Properties: &resourcepb.ResourceTableColumnDefinition_Properties{
+				Filterable: true,
+			},
+		},
+		{
+			Name:        USER_LOGIN,
+			Type:        resourcepb.ResourceTableColumnDefinition_STRING,
+			Description: "The login of the user",
+			Properties: &resourcepb.ResourceTableColumnDefinition_Properties{
+				UniqueValues: true,
+				Filterable:   true,
+			},
+		},
+	})
+	return resource.DocumentBuilderInfo{
+		GroupResource: iamv0.UserResourceInfo.GroupResource(),
+		Fields:        fields,
+		Builder:       new(userDocumentBuilder),
+	}, err
+}
+
+var _ resource.DocumentBuilder = new(userDocumentBuilder)
+
+type userDocumentBuilder struct{}
+
+// BuildDocument implements resource.DocumentBuilder.
+func (u *userDocumentBuilder) BuildDocument(ctx context.Context, key *resourcepb.ResourceKey, rv int64, value []byte) (*resource.IndexableDocument, error) {
+	tmp := &unstructured.Unstructured{}
```

Comment:
unstructured is fine -- or you could use the typed User so you don't need to cast the email and login